oauthproxy.go

```diff
@@ -961,15 +961,57 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 //
 //nolint:gosimple
```

Comment:
We can remove this `nolint` & TODO above now that you are adding more functionality 👍

Reply:
Removed